src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
OK, I've removed the check and documented this in 41b8c5b3a4c46d3c2beb1dae6e170c295c19b198